nym-node-status-api/nym-node-status-api/src/http/state.rs

```diff
@@ -285,24 +200,27 @@ impl HttpCache {
             .await
     }
 
-    pub async fn get_mixnode_stats(&self, db: &DbPool, offset: i64) -> Vec<DailyStats> {
+    pub async fn get_mixnode_stats(&self, db: &DbPool, offset: usize) -> Vec<DailyStats> {
         match self.mixstats.get(MIXSTATS_LIST_KEY).await {
             Some(guard) => {
                 let read_lock = guard.read().await;
-                read_lock.to_vec()
+                let mut stats = read_lock.to_vec();
+
+                stats.truncate(MIXNODE_STATS_HISTORY_DAYS + offset);
+                stats.into_iter().skip(offset).collect()
             }
             None => {
-                let new_node_stats = crate::db::queries::get_daily_stats(db, offset)
+                let mut new_node_stats = crate::db::queries::get_daily_stats(db, 0)
                     .await
-                    .unwrap_or_default();
-                // for every day that's missing, fill it with cached historic data
-                let mut mixnode_stats = self
-                    .mixnode_historic_daily_stats
-                    .merge_with_historic_stats(new_node_stats);
-                mixnode_stats.truncate(30);
-
-                self.upsert_mixnode_stats(mixnode_stats.clone()).await;
-                mixnode_stats
+                    .unwrap_or_default()
+                    .into_iter()
+                    .rev()
+                    .collect::<Vec<_>>();
+                // cache result without offset
+                self.upsert_mixnode_stats(new_node_stats.clone()).await;
+
+                new_node_stats.truncate(MIXNODE_STATS_HISTORY_DAYS + offset);
```

Comment:
why are you truncating it here rather than just getting correct number from db in the first place?